File: go/fiber-squirrel-example/api/jutsu/jutsu.go
package jutsu

import (
	"encoding/json"
	"fiber-squirrel-example/providers"
	"fiber-squirrel-example/types"
	"net/http"

	fiber "github.com/gofiber/fiber/v2"
)

func RegisterJutsu(app *fiber.App, providers *providers.Providers) {
	r := app.Group("/api/jutsu")
	{
		// Get jutsu
		r.Get("/:id/", func(c *fiber.Ctx) error {
			id := c.Params("id")
			jutsu, err := providers.PGDAL.GetJutsu(id)
			if err != nil {
				return c.SendStatus(http.StatusInternalServerError)
			}
			return c.JSON(jutsu)
		})

		// Insert jutsu
		r.Post("/", func(c *fiber.Ctx) error {
			var jutsuNew types.JutsuNew
			body := c.Body()
			err := json.Unmarshal(body, &jutsuNew)
			if err != nil {
				return c.SendStatus(http.StatusBadRequest)
			}
			jutsu, err := providers.PGDAL.CreateJutsu(jutsuNew)
			if err != nil {
				return c.SendStatus(http.StatusInternalServerError)
			}
			return c.JSON(jutsu)
		})

		// Update jutsu
		r.Put("/:id/", func(c *fiber.Ctx) error {
			id := c.Params("id")
			var jutsuUpdates types.JutsuNew
			body := c.Body()
			err := json.Unmarshal(body, &jutsuUpdates)
			if err != nil {
				return c.SendStatus(http.StatusBadRequest)
			}
			jutsu, err := providers.PGDAL.UpdateJutsu(id, jutsuUpdates)
			if err != nil {
				return c.SendStatus(http.StatusInternalServerError)
			}
			return c.JSON(jutsu)
		})

		// Delete Jutsu
		r.Delete("/:id/", func(c *fiber.Ctx) error {
			id := c.Params("id")
			jutsu, err := providers.PGDAL.DeleteJutsu(id)
			if err != nil {
				return c.SendStatus(http.StatusInternalServerError)
			}
			return c.JSON(jutsu)
		})
	}
}

File: py/fastapi-sqlalchemy-example/src/api/jutsu.py
"""Jutsu API routes."""

# pylint: disable=W0718

from uuid import UUID

from fastapi import APIRouter, FastAPI, Response  # Depends

from ..db.providers import Providers  # get_providers
from ..types.types import JutsuNew, JutsuUpdates


def add_jutsu_routes(app: FastAPI, providers: Providers):
    """Add jutsu routes."""
    router = APIRouter()

    @router.get("/{jutsu_id}")
    async def get_jutsu(
        response: Response,
        jutsu_id: str,
        # providers: Providers = Depends(get_providers),
    ):
        """Get jutsu."""
        try:
            jutsu = await providers.pgdal.get_jutsu(UUID(jutsu_id))
            response.status_code = 200
            return jutsu
        except Exception:
            response.status_code = 500
            return {}

    @router.post("/")
    async def create_jutsu(
        response: Response,
        jutsu_new: JutsuNew,
        # providers: Providers = Depends(get_providers),
    ):
        """Insert new jutsu."""
        try:
            jutsu = await providers.pgdal.create_jutsu(jutsu_new)
            response.status_code = 200
            return jutsu
        except Exception:
            response.status_code = 500
            return {}

    @router.put("/{jutsu_id}")
    async def update_jutsu(
        response: Response,
        jutsu_id: str,
        jutsu_updates: JutsuUpdates,
        # providers: Providers = Depends(get_providers),
    ):
        """Update existing jutsu."""
        try:
            jutsu = await providers.pgdal.update_jutsu(UUID(jutsu_id), jutsu_updates)
            response.status_code = 200
            return jutsu
        except Exception:
            response.status_code = 500
            return {}

    @router.delete("/{jutsu_id}")
    async def delete_jutsu(
        response: Response,
        jutsu_id: str,
        # providers: Providers = Depends(get_providers),
    ):
        """Delete existing jutsu."""
        try:
            jutsu = await providers.pgdal.delete_jutsu(UUID(jutsu_id))
            response.status_code = 200
            return jutsu
        except Exception:
            response.status_code = 500
            return {}

    app.include_router(
        router,
        prefix="/api/jutsu",
    )
